### DISCUSSION TITLE
How to avoid setting UIView width and height on native side?

### DISCUSSION THREAD

**Discussion by BraveEvidence** [reactwg/react-native-new-architecture#64] [Q&A]
I am trying to create a `fabric` module for `iOS`

In my `objective-c++` file while setting a `UIView`, I have to assign  a `CGRect` at time of `init` of `UIView`. If I don't on the native side and just give it on js side the view is not visible.

Following does not work

`Objective-C++ `
```
_view = [[UIView alloc] init];
_gradient = [CAGradientLayer layer];
_gradient.frame = _view.bounds;
    _gradient.startPoint = CGPointZero;
    _gradient.endPoint = CGPointMake(1, 1);
 _gradient.colors = @[(id)[UIColor redColor].CGColor,(id)[UIColor blackColor].CGColor,(id)[UIColor blueColor].CGColor];
    [_view.layer insertSublayer:_gradient atIndex:0];
self.contentView = _view;
```

`JS`

```
 <YourEdgeLinearGradientView
          style={{height: 200, width: 200, margin: 20}}
/>
```

Following works 

`Objective-C++ `
```
_view = [[UIView alloc] initWithFrame:CGRectMake(0,0,400,400)];
...
...
```

`JS`

```
 <YourEdgeLinearGradientView
          style={{height: 200, width: 200, margin: 20}}
/>
```

but the issue is it occupies the width and height set on the `native` side and ignores `js` side
I want to use
`_view = [[UIView alloc] init];`
without setting width and height on `native` side but setting it on `js` side

To add on more to this if I take a `UIButton` or a `UILAbel` instead of `CAGradientLayer` and apply constraints to `UIView` then I don't have to set `CGRectMake` to `UIView`

Also I don't want to pass width and height from the specs file, whatever I set from the style property should be applied

**Comment by sammy-SC** [comment]
Hello @PritishSawant,

I'm not sure what your end goal is, but if you create a view and set its frame manually, React Native will not know about that view and the view is under your control. React Native won't change size/location of view, it will only do so on view that it managers. 

If you are creating your own native component, you can set gradient directly on that component in native and let React Native figure out size and position of it.

I'm happy to look more in-depth into this if you provide more details and example for the issue.

**Comment by BraveEvidence** [comment]
[Check this for answer](https://stackoverflow.com/questions/73099414/how-to-avoid-setting-manual-uiview-width-and-height-on-native-side-for-fabric-co/73136807#73136807)
